### PR TITLE
FLUID-5708: Documentation related to switch component and adjuster

### PR DESCRIPTION
**Pull request by jobara**
https://issues.fluidproject.org/browse/FLUID-5708

Related Infusion PR: https://github.com/fluid-project/infusion/pull/811

### CODE REVIEW

File: src/documents/SwitchAPI.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+## Styling the Switch
+
+The Switch includes default CSS styles that it applies to its elements. The application of styles is based on known class names. The _default_ class names are described below, and can be used by including the Switch stylesheet:
+
+```html
+<link href="components/switch/css/Switch.css" type="text/css" rel="stylesheet" media="all">;
+```
+
+<div class="infusion-docs-note"><strong>Note:</strong> Switch.css is generated from the Switch.styl Stylus file.</div>
```
Comment (cindyli) [inline src/documents/SwitchAPI.md]: Would be helpful to add a link or include instructions of how to compile stylus into css.

File: src/documents/SwitchAPI.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+### Default Classes
+
+* `fl-switchUI` - Should be added to the switch container.
+* `fl-switchUI-text ` - Should be applied to the "on" and "off" indicators.
+* `fl-switchUI-control` - Should be applied to the actual switch control.
+* `fl-switchUI-controlKnob` - Should be applied to the knob of the switch control.
+
+## Dependencies
+
+The Image Reorderer dependencies can be met by including the `infusion-all.js` file in the header of the HTML file:
```
Comment (cindyli) [inline src/documents/SwitchAPI.md]: Fix the use of "The Image Reorderer dependencies ...".

File: src/documents/SwitchAPI.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+<script type="text/javascript" src="framework/core/js/Fluid.js"></script>
+<script type="text/javascript" src="framework/core/js/jquery.keyboard-a11y.js"></script>
+<script type="text/javascript" src="framework/core/js/FluidDocument.js"></script>
+<script type="text/javascript" src="framework/core/js/FluidIoC.js"></script>
+<script type="text/javascript" src="framework/core/js/DataBinding.js"></script>
+<script type="text/javascript" src="framework/core/js/FluidView.js"></script>
+<script type="text/javascript" src="components/switch/js/Switch.js"></script>
+```
+
+The Image Reorderer also requires the following stylesheet:
```
Comment (cindyli) [inline src/documents/SwitchAPI.md]: Same as above.

**Comment by jobara** [comment]
@cindyli thanks for the feedback, I've addressed your comments. Ready for more review.

File: src/documents/SwitchAPI.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+}
+                    </code>
+                </pre>
+            </td>
+        </tr>
+
+        <tr>
+            <td><code>model</code></td>
+            <td>The state of the switch.</td>
+            <td><code>true</code> is "on", <code>false</code> is off</td>
```
Comment (cindyli) [inline src/documents/SwitchAPI.md]: Missing quotes for the word `off` to match `on` in the first half of the sentence. 

Missing period at the end of the sentence.

File: src/documents/SwitchAPI.md [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                    </code>
+                </pre>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+## Styling the Switch
+
+The Switch includes default CSS styles that it applies to its elements. The application of styles is based on known class names. The _default_ class names are described below, and can be used by including the Switch stylesheet:
```
Comment (cindyli) [inline src/documents/SwitchAPI.md]: Remove `it` in the first sentence. Should be "... default CSS styles that applies ..."?

**Comment by cindyli** [comment]
The pull request that uses the new switch component in the preferences framework introduces a new css selector `fl-input-enhance` to have custom input controls work with the "inputs larger" preference. Shall the usage of this selector be documented in the preferences framework as well?

**Comment by jobara** [comment]
@cindyli and I talked about documenting `fl-input-enhance`. Currently this is just a new CSS style declaration in the Enactors.styl file. There are no specific code changes related to this. Additionally, there isn't currently any place in the existing documentation that seems to fit for adding this in. At some point we may want to document how all of the enactors work, and at that point this should be included in that documentation for input/button enhancement. For the time being we've decided not to add documentation for `fl-input-enhance`.

**Comment by cindyli** [comment]
Merged at 47114433f7ac15feafffc594080118e1f6732eae

**Pull request merged**